main.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'app'
+
+class LibraryApp
+  def initialize
+    @content = {
+      '1' => 'List all books',
+      '2' => 'List all people',
+      '3' => 'Create a person (teacher or student)',
+      '4' => 'Create a book',
+      '5' => 'Create a rental',
+      '6' => 'List all rentals for a given person id',
+      '7' => 'Exit'
+    }
+    @app = App.new
+  end
+
+  def welcome_message
+    puts 'Welcome to School library App!'
+  end
+
+  def display_menu
+    puts "\nPlease enter a number: "
+    @content.each { |index, string| puts "#{index} - #{string}" }
+  end
+
+  option = gets.chomp.to_i
+  case option
+  when 1
+    @app.list_books
+  when 2
+    @app.list_people
+  when 3
+    @app.create_person
+  when 4
+    @app.create_book
+  when 5
+    @app.create_rental
+  when 6
+    @app.list_rentals_by_person_id
+  when 7
+    puts 'You made a good choice'
+    exit
+  else
+    puts 'You are out of range'
+  end
+end
+
+def main
+  welcome_message
+  loop do
+    display_menu
+    option = gets.chomp.to_i
+    handle_option(option)
+  end
+end
```

Comment:
- [ ] Good job so far 👏👏 Overall, the code structure and logic seems fine. However, I encountered an issue while running the script using `ruby main.rb`. The program doesn't execute as expected, and there are no error messages displayed in the terminal⚠ and I'm unable to see the execution 😓 This is because when the code reaches line 26, it expects user input, `gets.chomp.to_i`, to select an option from the menu. However, at this point, the program has not yet reached any method that displays the menu. Therefore, the program waits for the user input without showing any menu or message

  ![image](https://github.com/tamoorsaeed22/OOP-school-library/assets/106119105/28f8157c-7efd-47aa-b054-e9f215de59de)


  1. The main issue with the code lies in the incorrect placement of the option processing code. The `case statement` and the code to process the options are placed outside any method within the `LibraryApp` class. Consequently, when the script is executed using `ruby main.rb`, the option processing code gets executed immediately before any method is defined, resulting in unexpected behavior. To fix this, we need to move the case statement and option processing logic inside a method

  
  2. Additionally, I noticed that the `handle_option` method is called in the `main` method, but it's not defined anywhere. We need to define this method within the `LibraryApp` class to handle the selected option properly.

_You can consider something like below 👇 Please test it before asking for another review_

```
 def handle_option(option)
    case option
    when 1
      @app.list_books
    when 2
      @app.list_people
    when 3
      @app.create_person
    when 4
      @app.create_book
    when 5
      @app.create_rental
    when 6
      @app.list_rentals_by_person_id
    when 7
      puts 'You made a good choice'
      exit
    else
      puts 'You are out of range'
    end
  end

  def main
    welcome_message
    loop do
      display_menu
      option = gets.chomp.to_i
      handle_option(option)
    end
  end
```

Reply:
@KanzaTahreem Hi, yes actually the problem with the handle_option was that it had Cyclomatic complexity for handle_option is a too high error which was not working, and after attempting multiple attempts I got to resolve that issue, thanks for correcting well function is working as required.